src/main/java/org/apache/mesos/specification/DefaultTaskSpecification.java

```diff
@@ -19,24 +21,33 @@
     private final Protos.CommandInfo commandInfo;
     private final Collection<ResourceSpecification> resourceSpecifications;
     private final Collection<VolumeSpecification> volumeSpecifications;
+    private final Optional<PlacementRuleGenerator> placementOptional;
 
     public static DefaultTaskSpecification create(Protos.TaskInfo taskInfo) throws InvalidTaskSpecificationException {
+        //TODO(nick): Is the original placement constraint needed here?
+        //  If this is just for relaunching a task in the same place then it's probably a moot point..
+        //  Otherwise we'll need to implement serializing/deserializing the configured placement
+        //  constraints in the TaskInfo itself. But from looking at the callers, it doesn't look like
```

Comment:
I'm curious about this, which are the callers that could potentially care about placement constraints?